### ISSUE TITLE
np.expand_dims(im_as_np, 0)

### ISSUE THREAD

**Issue by GarfieldF** [utkuozbulak/pytorch-custom-dataset-examples#8]
Because  torch.from_numpy(pic.transpose((2, 0, 1))),  we need set the axis=2  in the np.expand_dims().

When converts a PIL.Image (RGB) or numpy.ndarray (H x W x C) to a torch.FloatTensor of shape (C x H x W), numpy array shape is [28,28,1], and then the corresponding shape of torch is [1,28,28].

**Comment by utkuozbulak** [comment]
Hello,

I'm having hard time finding which line you are referring to. Where is the below code in the repository? 
`torch.from_numpy(pic.transpose((2, 0, 1)))`

**Comment by GarfieldF** [comment]
Apologies for the ambiguity. I mean "im_as_np = np.expand_dims(im_as_np, 0)"  should be np.expand_dims(-1), maybe. 
"torch.from_numpy(pic.transpose((2, 0, 1)))" is the reason for it.

**Comment by utkuozbulak** [comment]
Hey,

If you need to expand from the rightmost channel expand it with '-1'. The reason it is expanded with '0' is because the channel dimension is missing for grayscaled images.

**Comment by GarfieldF** [comment]
yes,I know what you means. But when converting numpy to torch, the image's shape should be(H x W x C), not (C x H x W).

**Comment by utkuozbulak** [comment]
No, in pytorch the shape of inputs is (Batch) x C x H x W. In tensorflow the shape of the input is H x W x C

**Comment by GarfieldF** [comment]
Oh, sorry. You're right! I went to check again.
This problem is encountered when using data = torchvision.transforms.ToTensor(data). But you use torch.from_numpy(data).
